lib/reference_counter_api.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/errors/api_error_handling"
+
+# Gets data from reference-counter Toolforge tool
+# https://toolsadmin.wikimedia.org/tools/id/reference-counter
+class ReferenceCounterApi
+  include ApiErrorHandling
+
+  TOOLFORGE_SERVER_URL = 'https://reference-counter.toolforge.org'
+
+  def self.valid_wiki?(wiki)
+    return wiki.project != 'wikidata'
+  end
+
+  def initialize(wiki, update_service = nil)
+    raise InvalidProjectError unless ReferenceCounterApi.valid_wiki?(wiki)
+    @project_code = wiki.project
+    @language_code = wiki.language
+    @update_service = update_service
+  end
+
+  # This is the main entry point.
+  # Given a revision ID, it retrieves the reference count from the
+  # reference-counter Toolforge API.
+  # If the API response is not 200 or an error occurs, it returns -1.
+  # Any encountered errors are logged in Sentry.
+  def get_number_of_references_from_revision_id(rev_id)
+    response = toolforge_server.get(references_query_url(rev_id))
+    parsed_response = Oj.load(response.body)
+    if response.status == 200
+      return parsed_response['num_ref']
+    else
+      # Log the error and return -1
+      Sentry.capture_message 'Non-200 response hitting references counter API',
+                             level: 'warning', extra: { project_code: @project_code,
+                             language_code: @language_code, rev_id:,
+                             status_code: response.status, content: parsed_response }
```

Comment:
Using `Sentry.capture_message` directly (instead of the `ApiErrorHandling` here), because I want to log these messages as warnings, not as errors. In fact, there is no error here, it's just the new API returning something different from 200. I would like to log this to monitor the new API is working fine. But I didn't find a way to do that through the `ApiErrorHandling` class, as it requires to put errors on the `TYPICAL_ERRORS` list to log something as warning.